patches/expo-notifications+0.28.1.patch

```diff
@@ -0,0 +1,325 @@
++    lastNotificationResponseBundle = NotificationSerializer.toResponseBundleFromExtras(extras)
++    sendEvent(NEW_RESPONSE_EVENT_NAME, lastNotificationResponseBundle)
++  }
++
+   /**
+    * Callback called when [NotificationManager] gets informed of the fact of message dropping.
+    * Emits a [MESSAGES_DELETED_EVENT_NAME] event.
+diff --git a/node_modules/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/interfaces/NotificationListener.java b/node_modules/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/interfaces/NotificationListener.java
+index 7b1add7..e22ec26 100644
+--- a/node_modules/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/interfaces/NotificationListener.java
++++ b/node_modules/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/interfaces/NotificationListener.java
+@@ -1,5 +1,7 @@
+ package expo.modules.notifications.notifications.interfaces;
+ 
++import android.os.Bundle;
++
+ import com.google.firebase.messaging.FirebaseMessagingService;
+ 
+ import expo.modules.notifications.notifications.model.Notification;
+@@ -28,6 +30,14 @@ public interface NotificationListener {
+     return false;
+   }
+ 
++  /**
++   * Callback called when notification response is received through package lifecycle listeners
++   *
++   * @param extras Bundle of extras from the lifecycle method
++   */
++  default void onNotificationResponseIntentReceived(Bundle extras) {
++  }
++
+   /**
+    * Callback called when some notifications are dropped.
+    * See {@link FirebaseMessagingService#onDeletedMessages()}
+diff --git a/node_modules/expo-notifications/android/src/main/java/expo/modules/notifications/service/delegates/ExpoNotificationLifecycleListener.java b/node_modules/expo-notifications/android/src/main/java/expo/modules/notifications/service/delegates/ExpoNotificationLifecycleListener.java
+new file mode 100644
+index 0000000..55e3ab9
+--- /dev/null
++++ b/node_modules/expo-notifications/android/src/main/java/expo/modules/notifications/service/delegates/ExpoNotificationLifecycleListener.java
+@@ -0,0 +1,79 @@
++package expo.modules.notifications.service.delegates;
++
++import android.app.Activity;
++import android.app.NotificationChannel;
++import android.app.PendingIntent;
++import android.content.Context;
++import android.content.Intent;
++import android.os.Bundle;
++import android.os.Parcel;
++import android.util.Log;
++
++import androidx.core.app.NotificationCompat;
++
++import expo.modules.core.interfaces.ReactActivityLifecycleListener;
++import expo.modules.notifications.notifications.NotificationManager;
++import expo.modules.notifications.notifications.model.Notification;
++import expo.modules.notifications.notifications.model.NotificationResponse;
++
++public class ExpoNotificationLifecycleListener implements ReactActivityLifecycleListener {
++
++    private Context mContext;
```

Comment:
Do we need to hold on to the `Context`?

Reply:
Good catch, I think this was automatically added in an Android Studio refactor. Will remove.

Reply:
Fixed